pkgs/development/python-modules/huawei-solar/default.nix

```
@@ -0,0 +1,44 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchFromGitLab,
+  home-assistant,
+}:
+let
+  python3 = home-assistant.python;
+in
+
+python3.pkgs.buildPythonPackage rec {
+  pname = "huawei-solar";
+  version = "2.3.0";
+  pyproject = true;
+
+  src = fetchFromGitLab {
+    owner = "Emilv2";
+    repo = pname;
+    rev = version;
+    hash = "sha256-PcpyyEH3Ad9oyr4aPlUgxU5S/NPoIDUZj+Ncs7FXhVA=";
+  };
+
+  build-system = with python3.pkgs; [
+    hatchling
+    hatch-vcs
+  ];
+  dependencies = with python3.pkgs; [
+    backoff
+    pytz
+    pymodbus
+    typing-extensions
+    pyserial-asyncio
+  ];
```

Consume the arguments in the function arguments. This is necessary so we get the correct version for the various versioned python package sets (python3{9,10,11,12,13,14}Packages).


```suggestion
{
  lib,
  buildPythonPackage
  fetchFromGitLab,
  hatchling
  hatch-vcs
  backoff
  pytz
  pymodbus
  pyserial-asyncio
  typing-extensions
}:

python3.pkgs.buildPythonPackage rec {
  pname = "huawei-solar";
  version = "2.3.0";
  pyproject = true;

  src = fetchFromGitLab {
    owner = "Emilv2";
    repo = pname;
    rev = version;
    hash = "sha256-PcpyyEH3Ad9oyr4aPlUgxU5S/NPoIDUZj+Ncs7FXhVA=";
  };

  build-system = [
    hatchling
    hatch-vcs
  ];

  dependencies = [
    backoff
    pytz
    pymodbus
    pyserial-asyncio
    typing-extensions
  ];
```

---

Done in latest commit, builds fine.